src/Workspaces/CSharp/Portable/Classification/Worker.cs

```diff
@@ -80,10 +80,28 @@ private void ClassifyNodeOrToken(SyntaxNodeOrToken nodeOrToken)
 
     private void ClassifyNode(SyntaxNode node)
     {
-        foreach (var token in node.DescendantTokens(span: _textSpan, descendIntoTrivia: false))
+        using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var stack);
+        stack.Push(node);
+
+        while (stack.TryPop(out var current))
         {
             _cancellationToken.ThrowIfCancellationRequested();
-            ClassifyToken(token);
+
+            // It's ok that we're not pushing in reverse.  The caller (TotalClassificationTaggerProvider) will be
+            // sorting the results before doing anything with them.
+            foreach (var child in current.ChildNodesAndTokens())
```

Comment:
>ChildNodesAndTokens [](http://example.com/codeflow?start=42&length=19)

this is way off topic, but I took another look at the SyntaxList.Enumerator's call to ItemInternal, and I'm not clear why the slotIndex isn't held between calls. By not doing so, the first loop in ItemInternal would need to walk the green slots until it reaches the appropriate slot index.

Again, off topic, so nothing needed in this PR, but it seems odd.

Reply:
def look into that.  there are some comments explaining things.  but it's possible they're stale, or there is room for improvement here.

Reply:
Note: i'm saying @ToddGrun should look into this :)